src/main/java/io/supertokens/webserver/api/useridmapping/UserIdMappingAPI.java

```diff
@@ -94,18 +95,16 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         }
 
         try {
-            AppIdentifierWithStorageAndUserIdMapping appIdentifierWithStorageAndUserIdMapping =
-                    this.getAppIdentifierWithStorageAndUserIdMappingFromRequest(req, superTokensUserId, UserIdType.SUPERTOKENS);
-
-            UserIdMapping.createUserIdMapping(main, appIdentifierWithStorageAndUserIdMapping.appIdentifierWithStorage,
+            UserIdMapping.createUserIdMapping(
+                    getAppIdentifier(req), enforcePublicTenantAndGetAllStoragesForApp(req),
```

Comment:
eve thought we give it all storages, it will still only add the mapping to the storage in which the user id is there right?

Reply:
yes